### DISCUSSION TITLE
Type for Flatten Error

### DISCUSSION THREAD

**Discussion by ksjitendra18** [open-circle/valibot#640] [Q&A] [answered]
Hey there. What should be the type for flat error. In this I am current manually typing it.

```ts
const [error, setError] = createSignal<FlatErrors<"name" | "email"> | null>(
  null,
);

if (result.success) {
  console.log(result.output);
} else {
  const err = flatten<typeof RegisterSchema>(result.issues);
  setError(err);
}
```

Here is the schema

```ts
const RegisterSchema = object({
  name: pipe(
    string(),
    nonEmpty("Please enter your name."),
    minLength(2, "Name should be 2 or more than 2 character"),
    maxLength(48, "Name should be less than"),
    trim(),
  ),
  email: pipe(
    string(),
    nonEmpty("Please enter your email."),
    email("Please enter a valid email address"),
  ),
});

export default RegisterSchema;

export type RegisterData = InferOutput<typeof RegisterSchema>;

```

Need help regarding this.

**Comment by fabian-hiller** [comment]
I agree that the current implementation is not perfect. I will probably change it back to `FlatErrors<typeof RegisterSchema>`. What do you think about that?

**Reply by ksjitendra18** [reply]
Yea It will be good if we could infer them. That would reduce the manual typing part and will ensure consistency.

**Reply by fabian-hiller** [reply]
Thank you for your feedback. This will be fixed in the next version.

**Reply by fabian-hiller** [reply, marked as answer]
A new version with the fix is available.

**Reply by ksjitendra18** [reply]
Thank you, this fixes the need of manual typing. Awesome work Fabian
